CMakeLists.txt

```diff
@@ -70,10 +71,46 @@ if(NATS_COVERAGE)
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage -fprofile-arcs -ftest-coverage")
 endif(NATS_COVERAGE)
 
-if(NATS_BUILD_WITH_TLS)
+if(NATS_BUILD_WITH_TLS AND NOT NATS_BUILD_WITH_WOLFSSL)
   set(OPENSSL_USE_STATIC_LIBS ${NATS_BUILD_OPENSSL_STATIC_LIBS})
   find_package(OpenSSL REQUIRED)
-endif(NATS_BUILD_WITH_TLS)
+endif(NATS_BUILD_WITH_TLS AND NOT NATS_BUILD_WITH_WOLFSSL)
+
+if(NATS_BUILD_WITH_WOLFSSL)
+  if(DEFINED ENV{NATS_WOLFSSL_DIR})
```

Comment:
All this is a bit complex, no? Look at what we do for OpenSSL: `find_package(OpenSSL REQUIRED)`. Is there no equivalent for that?